### ISSUE TITLE
Won't Import to Anki

### ISSUE THREAD

**Issue by HameerSS** [Garethp/RikaiRebuilt#2]
Upon pressing R, it says the card has been added.. but in reality there is no card.. I've tried reinstalling the addon in entirety, and tried both Chrome and Firefox.. nothing works.. please halp! <3 :'3

**Comment by HameerSS** [comment]
yessir!

**Comment by Garethp** [comment]
Can you paste what you put in the options for "Card Fields" in RikaiRebuilt and what your card layout looks like in Anki?

**Comment by HameerSS** [comment]
Hey Gareth buddy, thanks for reaching out, it works purrrrrfect!
I'll give you positive reviews now xD

Thanks again for all of this man! it really means a lot!! :')

**Comment by Garethp** [comment]
Ah, was it just misconfigured fields? I'm glad that's all it was. I'm really sorry that I didn't get around to helping you out earlier. Adding in EPWING took the time and energy I normally try to put in to supporting people

**Comment by HameerSS** [comment]
Lemme know where can I leave reviews, I can't find it.

**Comment by HameerSS** [comment]
It was not that, but it works now idk..

that must have been a task!!
are you in Japan bro?

**Comment by Garethp** [comment]
I actually don't know how to leave reviews either XD. I'd say on [this](https://addons.mozilla.org/en-US/firefox/addon/rikairebuilt/reviews/) page, but it's not showing me a button (probably because devs can't review their own plugins.

I'm in London, not Japan. I'm still learning Japanese myself. I ended up making this project because, like others, I really really didn't want to have to use a Firefox alternative just to use Rikaisama

**Comment by Garethp** [comment]
Oh, it looks like to leave a review you need to rate it out of stars on the [addon page](https://addons.mozilla.org/en-US/firefox/addon/rikairebuilt/), on the left

**Comment by HameerSS** [comment]
how do you this man! this is pure magic! xD
you working towards a JLPT or sth? *curious* I'm studying Japanese
full-time in my room outside of Japan.. it's been 5 years this year xD
but seriously, what programming languages allow you to make such a wonder?
R?

**Comment by Garethp** [comment]
It's been a really long process. For my job I work with JavaScript and PHP. But the JavaScript I usually write and the JavaScript that I need to use for this are two different things. So I started by basically copy-pasting code from the original plugin and cleaning up here and there, as well as looking up alternatives for what no longer exists (Mainly the stuff that makes the Anki Import and EPWING work). So JavaScript has been the *main* language here.

I've had to use Python to update the Real-Time Import plugin. More specifically, the plugin that came with the original Rikaisama communicated over UDP for some reason? And in a confusing format? So I copy-pasted some code from the Anki plugin for Yomichan to make it work like a normal thing. So there was a little bit of Python there, but not too much.

For EPWING, I took the original eplkup program source code (which was thankfully easy to get a hold of), and it's in C. I got the author (cb, the author of Rikaisama) to help me get set up on my computer too. Then I had to add in some code to make it communicate over JSON. And then make that compile in Windows and Linux. So more copy-pasting and Googling until things started working, then I started to improve the stuff I copy-pasted.

So the languages used so far: JavaScript, Python and C. How I did it: A lot of copy-pasting and then cleaning up.

As for Japanese, I'm working towards the N4 test that I've got booked in December. I haven't set the N4 as my goal, I just want to learn (and I've got a tutor who helps me with that over Discord), but I'll see in December if my learning so far compares to the N4. I went to Japan last year and could order food and ask where an ATM was and basic things in Japanese, so I'm super happy. I'm going again at the end of October and I'm hoping that I'll be able to form better sentences XD
